pkg/command/service/load.go

```diff
@@ -0,0 +1,423 @@
+	"io/ioutil"
+	"path/filepath"
+
+	"knative.dev/kperf/pkg/command/utils"
+
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"k8s.io/apimachinery/pkg/watch"
+
+	"knative.dev/kperf/pkg"
+	"knative.dev/serving/pkg/apis/serving"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+)
+
+const (
+	LoadOutputFilename = "ksvc_loading_time"
+)
+
+func NewServiceLoadCommand(p *pkg.PerfParams) *cobra.Command {
+	loadArgs := pkg.LoadArgs{}
+	serviceLoadCommand := &cobra.Command{
+		Use:   "load",
+		Short: "Load and Measure Knative service",
+		Long: `Scale Knative service from zero using load test tool and measure latency for service to scale form 0 to N
+
+For example:
+# To measure a Knative Service scaling from zero to N
+kperf service load --namespace ktest --svc-prefix ktest --range 0,3 --load-tool wrk --load-duration 60s --load-concurrency 40 --verbose --output /tmp`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Flags().NFlag() == 0 {
+				return fmt.Errorf("'service load' requires flag(s)")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return LoadServicesUpFromZero(p, loadArgs)
+		},
+	}
+
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.Namespace, "namespace", "", "", "Service namespace")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.NamespaceRange, "namespace-range", "", "", "Service namespace range")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.NamespacePrefix, "namespace-prefix", "", "", "Service namespace prefix")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.SvcPrefix, "svc-prefix", "", "", "Service name prefix")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.SvcRange, "range", "r", "", "Desired service range")
+	serviceLoadCommand.Flags().BoolVarP(&loadArgs.Verbose, "verbose", "v", false, "Service verbose result")
+	serviceLoadCommand.Flags().BoolVarP(&loadArgs.ResolvableDomain, "resolvable", "", false, "If Service endpoint resolvable url")
+	serviceLoadCommand.Flags().DurationVarP(&loadArgs.WaitPodsReadyDuration, "wait-time", "w", 10*time.Second, "Time to wait for all pods to be ready")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.LoadTool, "load-tool", "t", "wrk", "Select the load test tool, support wrk and hey")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.LoadConcurrency, "load-concurrency", "c", "30", "total number of workers to run concurrently for the load test tool")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.LoadDuration, "load-duration", "d", "60s", "Duration of the test for the load test tool")
+	serviceLoadCommand.Flags().StringVarP(&loadArgs.Output, "output", "o", ".", "Measure result location")
+
```

Comment:
I think we could extract some common flags of these commands, e.g. namespace, namespace-range, namespace-prefix, and create a method to add those shared flags. (possibly in a follow-on PR)